### PR TITLE
Close #198 Neurological disease related datasets added

### LINKED ISSUE #198: Feature Request:  Migrate 4 neurosciece specific datasets from admiralneuro to pharmaversesdtm
### Feature Idea

The following 4 neuroscience specific datasets created in admiralneuro. The package is in CRAN so we'd like to migrate the datasets to pharmaversesdtm.

`dm_neuro.rda`
`ag_neuro.rad`
`nv_neuro.rda`
`suppnv_neuro.rda`

### Relevant Input

- [x] - Copy programs from admiralneuro/data-raw/dm_neuro.R, ag_neuro.R, nv_neuro.R and suppnv_neuro.R
- [x] - Source the R codes to generate the corresponding .rda to the data/ folder.
- [x] - Reflect this update by adding the relevant information to the inst/extdata/sdtms-specs.json file.
- [x] - Run data-raw/create_sdtms_data.R in order to update NAMESPACE and update the .Rd files in man/.
- [x] - Add your GitHub handle to .github/CODEOWNERS.
- [x] - Update NEWS.md.

### PR DESCRIPTION
**Pull request by jwang-lilly**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jwang-lilly** [comment]
Please note `@importFrom packageName functionName` replaced library()

The following message shows up when run `lintr::lint_package()` if library() used
Switch to `@importFrom packageName functionName` resolves it.

Line 6 [undesirable_function_linter]
Function "library" is undesirable.
As an alternative, use roxygen2's @importFrom statement in packages
and `::` in scripts, instead of modifying the global search path. (edited)

**Comment by bundfussr** [comment]
> Please note `@importFrom packageName functionName` replaced library()
> 
> The following message shows up when run `lintr::lint_package()` if library() used Switch to `@importFrom packageName functionName` resolves it.
> 
> Line 6 [undesirable_function_linter] Function "library" is undesirable. As an alternative, use roxygen2's @importFrom statement in packages and `::` in scripts, instead of modifying the global search path. (edited)

The files in `data-raw` are not processed by `roxygen2`. Therefore roxygen tags like `@importFrom` doesn't have any effect. I would remove them.

It is fine to use undesirable functions like `library()` or `data()` in the scripts in `data-raw` because they won't be called by users. We have turned off the `undesirable_function_linter` for files in `data-raw` to avoid that such function calls are flagged as issue.

File: inst/extdata/sdtms-specs.json [hunk 1]
```diff
@@ -338,6 +338,30 @@
      "label": "Laboratory Measurements Dataset-metabolic",
      "description": "A SDTM LB dataset containing relevant laboratory measurements for metabolic studies",
      "source": "Constructed by {admiralmetabolic} developers"
+    },
+    {
+     "name": "dm_neuro",
+     "label": "Demographic Dataset-neuro",
+     "description": "A SDTM DM dataset for Alzheimer's disease studies",
+     "source": "Constructed by admiralneuro developers"
```
Comment (Fanny-Gautier) [inline inst/extdata/sdtms-specs.json]:
```suggestion
     "source": "Constructed by {admiralneuro} developers"
```

File: data-raw/ag_neuro.R [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Dataset: ag_neuro
+# Description: Create AG test SDTM dataset for Alzheimer's Disease (neuro studies)
+
```
Comment (Fanny-Gautier) [inline data-raw/ag_neuro.R]: Please remove unnecessary blank lines

File: data-raw/ag_neuro.R [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+library(admiral)
+library(dplyr)
+
+# Read input data ----
+
```
Comment (Fanny-Gautier) [inline data-raw/ag_neuro.R]: Please remove unnecessary blank lines

File: data-raw/nv_neuro.R [hunk 4]
```diff
@@ -0,0 +1,287 @@
+# Dataset: nv_neuro
+# Description: Create NV test SDTM dataset for Alzheimer's Disease (neuro studies)
+
+library(admiral)
+library(dplyr)
+library(lubridate)
+library(tibble)
+
+# Read input data ----
+
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/nv_neuro.R [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+visit_schedule <- pharmaversesdtm::vs |>
+  filter(USUBJID %in% dm_neuro$USUBJID) |>
+  filter(VISITNUM %in% c(3.0, 9.0, 13.0)) |>
+  rename(NVDTC = VSDTC, NVDY = VSDY) |>
+  select(USUBJID, VISITNUM, VISIT, VISITDY, NVDTC, NVDY) |>
+  group_by(USUBJID) |>
+  distinct()
+
+# All USUBJID have BASELINE but not all have visits 9 or 13 data
+
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/nv_neuro.R [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+  distinct() |>
+  unlist()
+
+visit13_usubjid <- visit_schedule |>
+  filter(VISITNUM == 13) |>
+  distinct() |>
+  unlist()
+
+# Create records for one USUBJID ----
+
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/nv_neuro.R [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+      paste(amy_tracer, "VISUAL CLASSIFICATION"),
+      paste(vendor, amy_tracer, "SUVR PIPELINE"),
+      paste(vendor, "FTP", "SUVR PIPELINE")
+    ),
+    VISITNUM = visitnum
+  )
+}
+
+# Create dataset for visit 3 (baseline) for all ids from dm_neuro ----
+
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/suppnv_neuro.R [hunk 8]
```diff
@@ -0,0 +1,54 @@
+# Dataset: suppnv_neuro
+# Description: Create SUPPNV test SDTM dataset for Alzheimer's Disease (neuro studies)
+
```
Comment (Fanny-Gautier) [inline data-raw/suppnv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/suppnv_neuro.R [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+library(admiral)
+library(dplyr)
+
+# Read input data ----
+
```
Comment (Fanny-Gautier) [inline data-raw/suppnv_neuro.R]: Please remove unnecessary blank lines

File: data-raw/ag_neuro.R (hunk 2 above)
Comment (Fanny-Gautier) [inline data-raw/ag_neuro.R]:
```suggestion

# Load libraries -----
```

File: data-raw/dm_neuro.R [hunk 10]
```diff
@@ -0,0 +1,65 @@
+# Dataset: dm_neuro
+# Description: Create DM test SDTM dataset for Alzheimer's Disease (Neuro)
+
```
Comment (Fanny-Gautier) [inline data-raw/dm_neuro.R]:
```suggestion

# Load libraries -----
```

File: data-raw/nv_neuro.R [hunk 11]
```diff
@@ -0,0 +1,287 @@
+# Dataset: nv_neuro
+# Description: Create NV test SDTM dataset for Alzheimer's Disease (neuro studies)
+
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R]:
```suggestion

# Load libraries -----
```

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
To maintain consistency with the rest of the package, please use the `{magrittr}` pipe `%>%` instead of the native pipe `|>`. Thank you.

File: data-raw/dm_neuro.R [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+library(admiral)
+library(dplyr)
+library(lubridate)
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/dm_neuro.R]: Not using haven in dm_neuro.R

**Comment by jwang-lilly** [comment]
@Fanny-Gautier, here is [tidyverse style guid](https://style.tidyverse.org/pipes.html#magrittr)
Base pipe provides all the features from magrittr that tidyverse recommends using base pipe instead of magrittr’s %>%.
